packages/core/src/components/Form.tsx

```diff
@@ -799,6 +818,18 @@ export default class Form<
     return this.getUsedFormData(formData, fieldNames);
   };
 
+  /** Allows a user to set a value for the provided `fieldPath`, which must be either a dotted path to the field OR a
+   * `FieldPathList`. To set the root element, used either `''` or `[]` for the path. Passing undefined will clear the
+   * value in the field.
+   *
+   * @param fieldPath - Either a dotted path to the field or the `FieldPathList` to the field
+   * @param [newValue] - The new value for the field
+   */
+  setFieldValue = (fieldPath: string | FieldPathList, newValue?: T) => {
+    const path = Array.isArray(fieldPath) ? fieldPath : fieldPath.split('.');
+    this.onChange(newValue, path);
```

Comment:
@nickgros I could also add the id on the the `onChange()` call by using `toFieldPathId()` and then passing `$id`... Thoughts about this?
```suggestion
    const fieldPathId = toFieldPathId('', registry.globalFormOptions, path);
    this.onChange(newValue, path, undefined, fieldPathId[ID_KEY]);
```